cmd/veneur-emit/main.go

```diff
@@ -1,29 +1,56 @@
 package main
 
 import (
+	"bytes"
 	"errors"
 	"flag"
 	"net"
 	"strings"
 	"time"
 
+	"fmt"
 	"github.com/DataDog/datadog-go/statsd"
 	"github.com/Sirupsen/logrus"
 	"github.com/gogo/protobuf/proto"
 	"github.com/stripe/veneur"
 	"github.com/stripe/veneur/ssf"
+	"strconv"
 )
 
 var (
+	// Generic flags
 	configFile = flag.String("f", "", "The Veneur config file to read for settings.")
 	hostport   = flag.String("hostport", "", "Hostname and port of destination. Must be used if config file is not present.")
-	name       = flag.String("name", "", "Name of metric to report. Ex: daemontools.service.starts")
-	gauge      = flag.Float64("gauge", 0, "Report a 'gauge' metric. Value must be float64.")
-	timing     = flag.Duration("timing", time.Now().Sub(time.Now()), "Report a 'timing' metric. Value must be parseable by time.ParseDuration (https://golang.org/pkg/time/#ParseDuration).")
-	count      = flag.Int64("count", 0, "Report a 'count' metric. Value must be an integer.")
-	tag        = flag.String("tag", "", "Tag(s) for metric, comma separated. Ex: service:airflow")
+	mode       = flag.String("mode", "metric", "Mode for veneur-emit. Must be one of: 'metric', 'event', 'sc'.")
 	debug      = flag.Bool("debug", false, "Turns on debug messages.")
-	toSSF      = flag.Bool("ssf", false, "Sends packets via SSF instead of StatsD. (https://github.com/stripe/veneur/blob/master/ssf/)")
+
+	// Metric flags
+	name   = flag.String("name", "", "Name of metric to report. Ex: 'daemontools.service.starts'")
+	gauge  = flag.Float64("gauge", 0, "Report a 'gauge' metric. Value must be float64.")
+	timing = flag.Duration("timing", 0*time.Millisecond, "Report a 'timing' metric. Value must be parseable by time.ParseDuration (https://golang.org/pkg/time/#ParseDuration).")
+	count  = flag.Int64("count", 0, "Report a 'count' metric. Value must be an integer.")
+	tag    = flag.String("tag", "", "Tag(s) for metric, comma separated. Ex: 'service:airflow'")
+	toSSF  = flag.Bool("ssf", false, "Sends packets via SSF instead of StatsD. (https://github.com/stripe/veneur/blob/master/ssf/)")
+
+	// Event flags
+	// TODO: what should flags be called?
```

Comment:
Ah. I don't have strong opinions here. Type 'em out and see what you think. I trust you. :)